DOC-849 Add benchmark processor to Self-Managed (#125)
redpanda-data/rp-connect-docs: modules/components/pages/processors/benchmark.adoc

= benchmark
// tag::single-source[]
:type: processor
:page-beta: true
:categories: ["Utility"]

component_type_dropdown::[]

> Logs throughput statistics for messages that pass through this processor, and provides a summary of those statistics over the lifetime of the processor.

Reviewer: ```suggestion
Logs message throughput statistics for this processor and summarizes them over its lifetime.
```

messages thought this processor seems redudant. Consider this simplification

Author: Generally, I'd avoid long noun strings. I've made some other updates. Thanks for the feedback.

Reviewer: Nice! I like the update

ifndef::env-cloud[]
Introduced in version 4.40.0.
endif::[]

```yml
# Configuration fields, showing default values
label: ""
benchmark:
  interval: 5s
  count_bytes: true
```

== Throughput statistics

This output logs the following rolling statistics at a <<interval,configurable interval>> to help you to understand the current performance of your pipeline:

- The number of messages processed per second.

> - The number of bytes of messages processed per second (optional).

Reviewer 2: "bytes of messages" feels weird to me here. I would just say "bytes processed per second"

Author: Updated.

For example:

```bash
INFO rolling stats: 1 msg/sec, 407 B/sec
```

> When the processor shuts down, it also writes a summary of the number and size of messages processed during the lifetime of the processor to the logs. For example:

Reviewer: ```suggestion
When the processor shuts down, it also logs a summary of the number and size of messages processed during its lifetime. For example:
```

Author: Updated.

```bash
INFO total stats: 1.00186 msg/sec, 425 B/sec 
```

== Fields

=== `interval`

How often to emit rolling statistics. Set to `0`, if you only want to log summary statistics when the processor shuts down.

*Type*: `string`

*Default*: `5s`

> === `count_bytes`

Reviewer 2: This processor has already been released with this name so we can't change it now without breaking the interface, but I will keep this in mind going forward.
